sqlmesh/core/engine_adapter/clickhouse.py

```diff
@@ -296,6 +293,70 @@ def _build_partitioned_by_exp(
             this=exp.Schema(expressions=partitioned_by),
         )
 
+    def inject_query_setting(
+        self,
+        query: Query,
+        setting_name: t.Optional[str] = "join_use_nulls",
+        setting_value: t.Optional[str] = "1",
+        check_server_default: bool = False,
+    ) -> Query:
+        # Set a setting in the query's SETTINGS clause
+        #
+        # Use in SCD models:
+        #  - The SCD query we build must include the setting `join_use_nulls = 1` to ensure that empty cells in a join
+        #      are filled with NULL instead of the default data type value. The default join_use_nulls value is `0`.
+        #  - The SCD embeds the user's original query in the `source` CTE
+        #  - Settings are dynamically scoped, so our setting may override the server's default setting the user expects
+        #      for their query.
+        #  - To prevent this, we:
+        #     - Determine if the user query already sets `join_use_nulls` - if so, we leave it as-is
+        #     - If the user query does not set `join_use_nulls`, we query the server for the current setting. If the
+        #         current setting is `0`, we add `join_use_nulls = 0` to the user query.
+        #  - We specify our setting on the outermost scope, so it applies to all CTEs/subqueries other than the user's
+        if setting_name and setting_value:
+            inject_setting = True
+            if check_server_default:
+                user_settings = query.args.get("settings")
+                if user_settings and [
+                    isinstance(setting, exp.EQ) and setting.this.this == setting_name
+                    for setting in user_settings
+                ]:
+                    inject_setting = False
+                else:
+                    server_setting = self.fetchone(
+                        f"SELECT value FROM system.settings WHERE name = '{setting_name}'"
+                    )[0]
+                    inject_setting = setting_value != server_setting
+                    setting_value = server_setting if inject_setting else setting_value
+
+            if inject_setting:
+                query.set(
+                    "settings",
+                    (query.args.get("settings") or [])
+                    + [
+                        exp.EQ(
+                            this=exp.var(setting_name),
+                            expression=exp.Literal(this=setting_value, is_string=False),
+                        )
+                    ],
+                )
```

Comment:
Consider refactoring this into:
```python
query.append("settings", exp.var(setting_name).eq_(exp.Literal.number(setting_value)))
```

Reply:
I didn't know about `append` - thank you!